### ISSUE TITLE
LayoutOptimizer optimizes to unsupported data_format for max_pool on CPU

### ISSUE THREAD

**Issue by FBerendsen** [tensorflow/tensorflow#23344]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):  v1.11.0-rc2-4-gc19e29306c 1.11.0
- Python version: 3.6.5
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 9.0.176/7.1.3 
- GPU model and memory: NVIDIA Quadro M2200

**Describe the current behavior**

TensorFlow automatically replaces my MaxPoolingOp by one using another data format which is not supported, subsequently. 
This throws the `InvalidArgumentError` as seen below.
The problem arose when upgrading from TensorFlow 1.8 to 1.11 and from the error it seems to be caused by a `TransposeNHWCToNCHW-LayoutOptimizer`. When isolating the issue to reproduce it, it seems that `max_pool`, `dataset` and `squeeze` are involved to raise the error.  The only related (closed) issue I could find: #19497 "NHWC convolution sometimes incorrectly considered NCHW" 

`
InvalidArgumentError (see above for traceback): Default MaxPoolingOp only supports NHWC on device type CPU
	 [[{{node label_image_dilated}} = MaxPool[T=DT_INT32, data_format="NCHW", ksize=[1, 1, 3, 3], padding="SAME", strides=[1, 1, 1, 1]](label_image_dilated-0-TransposeNHWCToNCHW-LayoutOptimizer)]]
	 [[{{node OneShotIterator_2}} = OneShotIterator[container="", dataset_factory=_make_dataset_UaZD9hBkHvg[], output_shapes=[[?,?]], output_types=[DT_INT32], shared_name="", _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]`

**Describe the expected behavior**

To not throw the error as was the case for TensorFlow 1.8.

**Code to reproduce the issue**
```python 
import tensorflow as tf
import numpy as np

dataset_np = {'height_or_width': 512, 
              'indices': np.random.randint(low=0, high=512, size=(1000,2),dtype=np.int64),
              'values' : np.random.randint(low=0, high=1000, size=(1000,),dtype=np.int32)}
dataset = tf.data.Dataset.from_tensors(dataset_np)

def densify(element):
    label_image_sparse = tf.SparseTensor(indices = element['indices'], 
                                         values = element['values'], 
                                         dense_shape = tf.cast(tf.stack([element['height_or_width'],
                                                                         element['height_or_width']]),tf.int64))
    label_image = tf.sparse_tensor_to_dense(label_image_sparse, 
                                            default_value=-1, 
                                            validate_indices=False, 
                                            name='label_image')
    label_image_dilated = tf.squeeze(tf.nn.max_pool([tf.expand_dims(label_image, axis=-1)], 
                                                     data_format="NHWC", 
                                                     ksize= [1,3,3,1], 
                                                     strides = [1,1,1,1], 
                                                     padding='SAME', 
                                                     name='label_image_dilated'),[0,-1])
    return {'label_image_dilated':label_image_dilated}

dataset = dataset.map(densify)
element = dataset.make_one_shot_iterator().get_next()

with tf.Session() as sess:
    result = sess.run(element)
    print(result)
```

**Other info / logs**

**Comment by ymodak** [comment]
I was able to execute your code successfully using TF 1.10, 1.11, and 1.12. 
Can you try doing fresh installation of TF and check again?

**Comment by FBerendsen** [comment]
On the computer of a colleague we installed tensorflow 1.12 from scratch and my code runs successfully indeed. It is a machine with Windows 10, but with different Nvidia gpu (Quadro K610M). I'll do a more thorough reinstall of the tensorflow dependency chain on my machine and will let you know.

**Comment by FBerendsen** [comment]
Still no success on my machine. I tried to do a rigorous reinstall, in short: 
  - Old setup: Python and CUDA via windows installers and tensorflow-gpu via pip. I removed all those, (including left over environment paths, files of cudnn and all python site-packages). 
   - New setup: I updated NVIDIA graphics driver and installed miniconda. 

Using different conda environments, my code works in TF 1.8, but in TF 1.9 and 1.11 still not. 
- If this is an installation issue, what files could there be still on my system?
- Could this be a hardware issue? Can my error be reproduced on a Quadro M2200? @ymodak I guess you used a different GPU?

**Comment by ymodak** [comment]
My GPU is Quadro K1200. Its strange that you are still having the issue after fresh install. I think you can try posting your issue on [Stack OverFlow](http://stackoverflow.com/questions/tagged/tensorflow) as well if haven't already.

**Comment by ymodak** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by FBerendsen** [comment]
For reference, #22145 discussed the very same problem. The proposed workaround I tested and worked for me: adding `with tf.device('/cpu:0'):` to the `densify` function.

**Comment by JanzenLiu** [comment]
Hey, I notice that your GPU is NVIDIA Quadro M2200. Do you remember how you set up the tensorflow environment? I kept getting  CUDA_ERROR_NO_DEVICE. My GPU is alos Quadro M2200.
